### ISSUE TITLE
API for single perspectives

### ISSUE THREAD

**Issue by bwesterb** [open-mpic/draft-mpic#15]
Wouldn't it be more flexible to use a model like with Certificate Transparency? Where a number of organizations could each run one perspective and CAs select the required number of perspectives?

_Originally posted by @romanf in https://github.com/open-mpic/draft-mpic/issues/8#issuecomment-2333957238_

**Comment by bwesterb** [comment]
This is an interesting suggestion. This is more complicated to use for a client than the current MPIC API, but has the potential to have a broader ecosystem.

The crux is: are there many parties that are not willing to run a global multi-perspective network, but are willing to run single perspectives. And are these parties sufficiently globally distributed?

**Comment by romanf** [comment]
I can answer for our org... we would be willing to run one perspective (probably redundant from our datacenters which are both located in Switzerland)... since that would basically not cause any additional cost.
Running a multi-perspective network would not cuase extra cost for big organisations (e.g. Google, Amazon, Microsoft, Cloudflare....) that have the gloal network already. But I'm not sure if they would be willing to offer their perspectives to CAs for free... ;)

**Comment by bwesterb** [comment]
I can't speak for the other large providers, but Cloudflare [has committed](https://blog.cloudflare.com/secure-certificate-issuance/) to offer the service for free.

**Comment by TheEnbyperor** [comment]
Speaking with my Glauca Digital hat on we'd also be willing to offer MPIC endpoints for one or two datacenters we have in Europe, but it wouldn't be feasible for us to offer a global service.

**Comment by birgelee** [comment]
I am happy this question came up in discussion. This actually reminds me of something Ryan Hurst (https://www.linkedin.com/in/ryanmhurst) had discussed with me where CAs assist other CAs with DCV.

In general I like the idea of moving in this direction, but it would likely require a different standard in addition to this proposed draft. I think this draft should be kept explicitly for the interface between CA infrastructure and an MPIC service. I feel standardizing the communication between an MPIC service and multiple perspectives run by different 3rd-party organizations is a good idea but deserves a separate standards draft.

Even prior to standardization, we may be able to offer something along these lines in the Open MPIC implementation.

**Comment by sciros** [comment]
Yes, from my point of view, this is a different paradigm from the single-endpoint MPIC approach and would require different endpoint definition(s).

The current open-mpic specification describes a classic client-server topology, i.e. a service which abstracts away the entire orchestration of MPIC, including how remote perspectives are interrogated and how quorum is determined. The endpoint of interest is client-to-orchestrator. (The orchestrator-to-remote-perspective integration is in this case not specified.)

The peer-to-peer model @romanf describes would have each peer carry our their own orchestration, so the endpoint specifications involved would be orchestrator-to-remote-perspective, rather than client-to-orchestrator.

Both models are valid per se, and presumably for some CAs one is a better fit than the other. However I believe the assumptions that underlie each, and the practical challenges inherent in each, are materially distinct enough that while one solution can _inform_ the other, I don't think one renders the other obsolete nor can one subsume the other.

**Comment by SulemanAhmadd** [comment]
Note that there are two different approaches: 1) CA selects its preference of single perspectives to use (changes the current API completely), or 2) The MPIC service (orchestrator) does it for the client transparently, similar to what we are doing right now but for multiple perspectives under the same org.

The former is obviously harder for the client. For the latter, yes, we have not defined any perspective <-> perspective API interface in the draft, but should we consider that strictly out of scope? Surely something to explain in the operations section of the draft. If we do, whether each of those perspectives belongs to the same organization or different organizations, the API interface can be made to remain consistent. It means more state management for the orchestrating MPIC service for sure i.e. if availability of compliant perspectives under org are less than quorum then reach out to other available perspectives from other orgs (easy on paper but can be difficult to coordinate the selection mechanism).

Regardless, one of the biggest challenge with the single perspective model is determining whether selection of third-party perspectives are sufficiently distributed to meet the distance requirements. 1, 2 or multiple perspectives, the problem here can be generalized to cross organization MPIC.
